### ISSUE TITLE
I am not getting relevant results  with m2-bert-80M-32k-retrieval

### ISSUE THREAD

**Issue by legaltextai** [HazyResearch/m2#38]
Embedded using the API 
Significantly underperforms other models 
In most of the cases, each embedding is a full text of the Supreme Court decision 
Indexed with hnsw. 
Should I use a different index? 
[togethercomputer/m2-bert-80M-32k-retrieval ](https://helpmefindcase.com/.7._embed:_m2-bert-80M-32k)
Thanks

**Comment by DanFu09** [comment]
CC @jonsaadfalcon 

Try the V1 models: https://huggingface.co/hazyresearch/M2-BERT-32K-Retrieval-Encoder-V1

Those have seen some legal data during training so hopefully they should work a bit better :)

If they still don't work, would love to hear more about the setup and see if there's a mismatch with how we trained them!

**Comment by legaltextai** [comment]
both query and the documents use the same embedding protocol, correct? i don't need to add any extra when embedding the prompt, like in UAE Large, right? OK to use cosine similarity and hnsw index?

**Comment by DanFu09** [comment]
Yes, both queries and documents use the same protocol and model, there's no extra prompt.

**Comment by Huguet57** [comment]
@legaltextai check that the `padding` is at `max_length` as seen in #43
